prover/src/lib.rs

```diff
@@ -257,15 +253,12 @@ pub trait Prover {
         // 1 ----- Commit to the execution trace --------------------------------------------------
 
         // build computation domain; this is used later for polynomial evaluations
-        #[cfg(feature = "std")]
-        let now = Instant::now();
-        let domain = StarkDomain::new(&air);
-        #[cfg(feature = "std")]
-        debug!(
-            "Built domain of 2^{} elements in {} ms",
-            domain.lde_domain_size().ilog2(),
-            now.elapsed().as_millis()
-        );
+        let lde_domain_size = air.lde_domain_size();
+        let domain = {
+            let _ = info_span!("build_domain", domain_size = lde_domain_size).entered();
+            StarkDomain::new(&air)
+        };
```

Comment:
It doesn't look like the approach I suggested works correctly (the times reported are way too small and don't add up to the final time). To make it work we'd need to do something like this:
```Rust
let lde_domain_size = air.lde_domain_size();
let domain = {
    let span = info_span!("build_domain", domain_size = lde_domain_size).entered();
    let domain = StarkDomain::new(&air);
    drop(span);
    domain
};
```
This may be fine for some other steps below (let's update them to make sure the times are captured correctly), but for this steps `.in_scope()` would probably be a better solution.

Also, a couple of other comments:
- Let's rename the field to `lde_domain_size`.
- Let's add `trace_length` field (this can also come from `air` struct).